CMakeLists.txt

```diff
@@ -63,14 +63,37 @@ if (NOT ZLIB_FOUND)
     endif()
 endif()
 
-find_package(GMP)
-if (NOT GMP_FOUND)
-    if (APPLE)
-        message(FATAL_ERROR "GMP not found. Try 'brew install gmp'.")
-    elseif (UNIX)
-        message(FATAL_ERROR "GMP not found. Try 'sudo apt install libgmp-dev' or 'sudo dnf install gmp-devel'.")
-    else()
-        message(FATAL_ERROR "GMP not found.")
+option(DOWNLOAD_GMP "Download libgmp and build the library locally instead of using a system installation" OFF)
+if (DOWNLOAD_GMP)
+    include(ExternalProject)
+    ExternalProject_Add(
+        gmp
+        URL https://gmplib.org/download/gmp/gmp-6.3.0.tar.xz
+        INSTALL_DIR ${CMAKE_CURRENT_BINARY_DIR}/gmp
+        CONFIGURE_COMMAND "<SOURCE_DIR>/configure" "--prefix=<INSTALL_DIR>" --disable-shared --enable-static
+        USES_TERMINAL_CONFIGURE TRUE
+        BUILD_IN_SOURCE FALSE
```

Comment:
Presumably `BUILD_IN_SOURCE FALSE` is the default?

Reply:
Ah yes, that is a leftover from when I set it true before I realized I could use "<SOURCE_DIR>/configure" as the configure command.
```suggestion
```